tensorflow_lite_support/ios/task/vision/utils/sources/GMLImageUtils.m

```diff
@@ -0,0 +1,305 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+#import "tensorflow_lite_support/ios/task/vision/utils/sources/GMLImageUtils.h"
+#import "tensorflow_lite_support/ios/sources/TFLCommon.h"
+#import "tensorflow_lite_support/ios/sources/TFLCommonUtils.h"
+
+#include "tensorflow_lite_support/c/task/vision/core/frame_buffer.h"
+
+@import CoreVideo;
+@import Accelerate;
+@import CoreImage;
+@import CoreGraphics;
+
+@interface TFLCVPixelBufferUtils : NSObject
++ (uint8_t *_Nullable)convertBGRAtoRGBforPixelBufferBaseAddress:(CVPixelBufferRef)pixelBuffer
+                                                          error:(NSError **)error;
+@end
+
+@interface UIImage (RawPixelDataUtils)
+- (TfLiteFrameBuffer *)frameBufferWithError:(NSError **)error;
+@end
+
+@implementation TFLCVPixelBufferUtils
+
++ (uint8_t *)convertBGRAtoRGBforPixelBufferBaseAddress:(CVPixelBufferRef)pixelBuffer
+                                                 error:(NSError **)error {
+  size_t width = CVPixelBufferGetWidth(pixelBuffer);
+  size_t height = CVPixelBufferGetHeight(pixelBuffer);
+  size_t stride = CVPixelBufferGetBytesPerRow(pixelBuffer);
+
+  int destinationChannelCount = 3;
+  size_t destinationBytesPerRow = destinationChannelCount * width;
+
+  uint8_t *pixelBufferBaseAddress = (uint8_t *)CVPixelBufferGetBaseAddress(pixelBuffer);
+
+  uint8_t *destPixelBufferAddress = [TFLCommonUtils mallocWithSize:height * destinationBytesPerRow
+                                                             error:error];
+
+  if (!destPixelBufferAddress) {
+    return NULL;
+  }
+
+  vImage_Buffer srcBuffer = {
+      .data = pixelBufferBaseAddress, .height = height, .width = width, .rowBytes = stride};
+
+  vImage_Buffer destBuffer = {.data = destPixelBufferAddress,
+                              .height = height,
+                              .width = width,
+                              .rowBytes = destinationBytesPerRow};
+
+  vImage_Error convertError = kvImageNoError;
+  convertError = vImageConvert_BGRA8888toRGB888(&srcBuffer, &destBuffer, kvImageNoFlags);
+
+  if (convertError != kvImageNoError) {
+    [TFLCommonUtils customErrorWithCode:TFLSupportErrorCodeImageProcessingError
+                            description:@"Image Format Conversion Failed."
```

Comment:
nit: Let's unify the format of description to a normal sentence case. (e.g. Image format conversion failed.)